core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -268,32 +269,31 @@ object Executor {
 
   private[caliban] def mergeFields(field: Field, typeName: String): List[Field] = {
     // ugly mutable code but it's worth it for the speed ;)
-    val array = ArrayBuffer.empty[Field]
-    val map   = collection.mutable.Map.empty[String, Int]
-    var index = 0
+    val map      = new java.util.LinkedHashMap[String, Field]()
```

Comment:
Did you measure this change?

Reply:
I didn't measure it this time around, but I did when I changed the same logic in `caliban.execution.Field`. The performance is the same with both implementations, with the difference that using a `LinkedHashMap` is cleaner. Resources online seem to suggest that the performance of `LinkedHashMap` and `HashMap` should be roughly identical as well